components/automate-ui/src/app/pages/integrations/aws-form/aws-form.component.ts

```diff
@@ -1,19 +1,11 @@
-import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
-import { FormGroup } from '@angular/forms';
-import { get } from 'lodash/fp';
+import { Component, ChangeDetectionStrategy } from '@angular/core';
+import { BaseIntegrationFormComponent } from 'app/pages/integrations/base-integration-form.component';
 
 @Component({
   selector: 'app-aws-integration-form',
   templateUrl: './aws-form.component.html',
   styleUrls: ['./aws-form.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class IntegrationsAWSFormComponent {
-  @Input() formGroup: FormGroup;
-
-  showInstanceCreds() {
-    const formData = this.formGroup.value;
-    const no_creds = get('no_creds', formData);
-    return !no_creds;
-  }
```

Comment:
I put all this replicated code in a common base class.

Reply:
I kind of prefer simple duplication like this over adding extension classes but that's just my preference.

Reply:
OK, my preference is rather weak, so I will revert that.